File: Dataset/cpp/months.cpp

//  months.cpp (c) 1998-2002 Kari Laitinen

#include  <iostream.h>

int main()
{
   char*  names_of_months[]  =

      { "January", "February", "March", "April", "May", "June",
        "July", "August", "September", "October", "November",
        "December"  }  ;

   cout <<  "\n The first month of year is "
        <<  names_of_months[ 0 ]  <<  "." ;

   cout <<  "\n\n The seventh month, " << names_of_months[ 6 ]
        <<  ", is named after Julius Caesar.\n" ;

   cout << hex << "\n Let's explore the memory: \n" ;

   for ( int month_index  =  0 ;
             month_index  <  4 ;
             month_index  ++  )
   {
      cout<< "\n Address " << (long) &names_of_months[ month_index ]
          << " contains "  << (long)  names_of_months[ month_index ]
          << " (Address of \""  <<    names_of_months[ month_index ]
          << "\")" ;
   }

   cout  <<  "\n" ;
   char*  memory_pointer  =  names_of_months[ 0 ] ;

   while ( *memory_pointer != 'b' )
   {
      cout << "\n Address " << (long) memory_pointer 
           << " contains  " << (int) *memory_pointer
           << "   "         <<       *memory_pointer ;
      memory_pointer  ++ ;
   }
}

File: Dataset/cs/Months.cs

//  Months.cs  (c) 2004 Kari Laitinen

//  www.naturalprogramming.com

//  If you are a reader who also studies my C++ book,
//  please note that this program does not work exactly
//  the same way as months.cpp in the C++ book.
//  The C++ program uses pointers, and pointers are not
//  widely discussed in the C# book.

using System ;

class Months
{
   static void Main()
   {
      string[]  names_of_months  =

         { "January", "February", "March", "April", "May", "June",
           "July", "August", "September", "October", "November",
           "December"  }  ;

      Console.Write( "\n The first month of year is "
                  +  names_of_months[ 0 ]  +  "." ) ;

      Console.Write(  "\n\n The seventh month, " + names_of_months[ 6 ]
                  +  ", is named after Julius Caesar.\n" ) ;

      Console.Write( "\n Our calendar has " +  names_of_months.Length
                  +  " months. \n" ) ;

      for ( int month_index  =  0 ;
                month_index  <  4 ;
                month_index  ++  )
      {
         int number_of_letters_in_month_name  =
                             names_of_months[ month_index ].Length ;

         Console.Write( "\n " +  names_of_months[ month_index ]
                     +  " is made of "
                     +  number_of_letters_in_month_name
                     +  " letters: " ) ;

         for ( int letter_index  =  0 ;
                   letter_index  <  number_of_letters_in_month_name ;
                   letter_index  ++ )
         {
            Console.Write(  " "  +
                  names_of_months[ month_index ] [ letter_index ] ) ;
         }            
      }
   }
}

/* Multiline comment begins:

   The following array is to be used in an exercise related
   to Months.cs:

      string[]  history_of_months =
      {  "month of Roman god Janus",      // January
         "last month in Roman calendar",  // February
         "month of Roman war god Mars",   // March
         "month of Roman goddess Venus",  // April
         "month of goddess Maia",         // May
         "month of Roman goddess Juno",   // June
         "month of Julius Caesar",        // July
         "month of Emperor Augustus",     // August
         "7th Roman month",               // September
         "8th Roman month",               // October
         "9th Roman month",               // November
         "10th Roman month"  } ;          // December

   Mulitiline comment ends.  */






